op-service/eth/types.go

```diff
@@ -619,3 +620,41 @@ const (
 	GetPayloadV2 EngineAPIMethod = "engine_getPayloadV2"
 	GetPayloadV3 EngineAPIMethod = "engine_getPayloadV3"
 )
+
+// StorageKey is a marshaling utility for hex-encoded storage keys, which can have leading 0s and are
+// an arbitrary length.
+type StorageKey []byte
+
+func (k *StorageKey) UnmarshalJSON(text []byte) error {
```

Comment:
If you implement the `UnmarshalText(data []byte) error` interface instead, the JSON lib will use that, and you won't need the `json.Unmarshal` into a `string`.

Reply:
nice simplification! just updated the PR